web-marketplace/src/hooks/projects/useProjectsWithOrders.ts

```diff
@@ -35,6 +38,7 @@ export interface ProjectsWithOrdersProps {
   projectId?: string; // to filter by project
   skippedProjectId?: string; // to discard a specific project
   classId?: string; // to filter by class
+  pinnedIds?: string[]; // list of on-chain id, uuid or slug to pinned at the top
```

Comment:
I don't see the pinned project if I provide a slug or uuid, looks like it might not be handled yet here?

Reply:
It's most likely because the project you tried to pin was not approved or published. We're using the [following hook](https://github.com/regen-network/regen-web/blob/dev/web-marketplace/src/pages/Projects/hooks/useOffChainProjects.tsx#L31) to get off-chain projects there.
Also note that slug take precedent over uuid: https://github.com/regen-network/regen-web/blob/dev/web-marketplace/src/lib/normalizers/projects/normalizeProjectsWithMetadata.ts#L85-L90
Could probably fix that by adding a `uuid` or `offchainId` in the normalizedObject.

Reply:
I've tried with "woodburn" as slug which is both published and approved

```psql
postgres=> select approved, published from project where slug='woodburn';
 approved | published 
----------+-----------
 t        | t
```

Reply:
Made a fix for the last part mentioned above here: [0ac4071](https://github.com/regen-network/regen-web/pull/2191/commits/0ac4071c84fe046c37f0cf09063234580780ab8d)

Reply:
@blushi There an in the commit I made earlier, which I just fixed. Also I stopped relying on the `name` field and created separate ones for both `uuid` and `slug` this should unsure consistent results. Woodburn is now correctly pinned here: https://deploy-preview-2191--regen-marketplace.netlify.app/

Reply:
Working now for woodburn!
Just a nit but would it be possible to open up the project page using the slug too? currently it uses the uuid which might not be the best for sharing.
But I still don't see the 2nd featured project with a uuid, although it's published and approved:

![image](https://github.com/regen-network/regen-web/assets/3149358/6b4c90ad-31b3-4d0d-862f-46ee232f931c)

Reply:
oh I guess that might be due to the projects on chain pagination limit...

Reply:
It's because this project is neither returned by the on-chain projects hooks (beyond the 100 projects current limitation) or the off-chain projects hooks (since it has an on-chain id).

Reply:
yeah I figured, I'll create an issue to track the on chain queries pagination default limit related work, I guess one solution would be to query for all the pages to make sure we get all projects or credit classes...

Reply:
I've made the modification to use the slug whenever available here: [c64154f](https://github.com/regen-network/regen-web/pull/2191/commits/c64154fbf658e384a355beefc708df04cca300d4)